### PR TITLE
Add SelectInsideTags and SelectAroundTags text-objects

### PR DESCRIPTION
**Pull request by MarkusSN**
This adds support for the tags text objects, but currently only works for single line tags. I am planning on adding support for multiline tags, but I decided some early feedback would be useful.

### CODE REVIEW

**Comment by MarkusSN** [comment]
Added support for multiline tags.

**Comment by MarkusSN** [comment]
#598

File: spec/text-objects-spec.coffee [hunk 1]
```diff
@@ -164,28 +164,232 @@ describe "TextObjects", ->
...
+
     beforeEach ->
-      editor.setText("<something>here</something><again>")
-      editor.setCursorScreenPosition([0, 5])
+      editor.setText('clean')
+      editor.setCursorBufferPosition([0, 0])
+      keydown('y')
+      keydown('i')
+      keydown('w')
+      expect(vimState.getRegister('"').text).toBe('clean')
```
Comment (bronson) [inline spec/text-objects-spec.coffee]: I don't understand this...?  Looks like you're testing yanking inside a word before each inner tag test?

**Comment by MarkusSN** [comment]
Lol, I understand the confusion. I just wanted to make sure that I reset the register before each test, and that was the first thing I came up with. The plan was to find a better way to set the value in the register, but I obviously forgot :)

File: spec/text-objects-spec.coffee (hunk 1 above)
Comment (MarkusSN) [inline spec/text-objects-spec.coffee]: Lol, I understand the confusion. I just wanted to make sure that I reset the register before each test, and that was the first thing I came up with. The plan was to find a better way to set the value in the register, but I obviously forgot :)

**Comment by MarkusSN** [comment]
Currently, the implementation assumes a correctly formatted xml structure. But it seems like unclosed tags are ignored in vim, so I am going to try to get that working.

**Comment by MarkusSN** [comment]
Rebased.
Seems to working pretty well now. It's a bit massive compared to some of the other text objects, so if anyone has any feedback, that would be great :)

**Comment by jacekkopecky** [comment]
Would this address #835?

**Comment by MarkusSN** [comment]
Yes :)

**Comment by MarkusSN** [comment]
I changed one of the existing tests to expect that "it does _not expand_ an existing selection in visual mode" instead of "it expands an existing selection in visual mode". This is more in line with how it works in vim.

**Comment by lee-dohm** [comment]
As stated in [the README](https://github.com/atom/vim-mode), this package is no longer maintained and is deprecated. We recommend that people use the [vim-mode-plus package](https://atom.io/packages/vim-mode-plus) instead. Because of this, we are [archiving this repository](https://help.github.com/articles/about-archiving-repositories/) and closing all issues and pull requests. Thanks very much for your support and contributions!

**Pull request closed without merging**